### DISCUSSION TITLE
Can't find error log after running `qlty init`

### DISCUSSION THREAD

**Discussion by nk-ty** [qltysh/qlty#1256] [Q&A]
### Question

When running `qlty init` in the following environment, a ruby ​​installation error occurs.
The error log file that is output at that time cannot be found.

Environment:
- macoS 13.3.1
- Intel chip
- ruby ​​3.3.0 (2023-12-25 revision 5124f9ac75) [x86_64-darwin22]

### How to reproduce (optional)

1. Clone any repository
2. Go to path to local repository
3. Run `qlty init`
4. An error occurs

```bash
git clone git@github.com:lifull-dev/bucky-management.git
cd bucky-management

qlty init
› Initializing qlty in the current directory...
✔ Created .qlty/ directory
✔ Fetched plugins definitions
✔ Created .gitignore in .qlty/
✔ Created qlty.toml config file with 13 plugins

Plugin         Version  Targets   Config
trivy          latest   17 files  
osv-scanner    latest   1 files   
yamllint       latest   14 files  
checkov        latest   16 files  
eslint         8.57.0   7 files   
ripgrep        latest   36 files  
stylelint      14.6.1   3 files   
brakeman       latest   27 files  
rubocop        1.65.1   27 files  
prettier       latest   28 files  
markdownlint   0.41.0   1 files   
golangci-lint  latest   0 files   
trufflehog     latest   96 files  

✔ Set up qlty in the current directory
✔ Would you like to run plugins to see a sample of issues? · yes

› Running qlty check --sample 5

     [0/1] 🤔  Planning...  0.14s
     [1/1] 🔍  Analyzing 5 samples... 
❌ Error installing ruby@3.3.0.

    See more: /Users/nk/.qlty/cache/tools/ruby/3.3.0-ce8ae37c7b0b-install.log

Caused by:
    Failed to get version for package "ruby": (command sh -c "ruby --version" exited with code 0)
❌ command ["/Users/nk/.qlty/bin/qlty", "check", "--sample", "5"] exited with code 99

ls -ls /Users/nk/.qlty/cache/tools/ruby/3.3.0-ce8ae37c7b0b-install.log
ls: /Users/nk/.qlty/cache/tools/ruby/3.3.0-ce8ae37c7b0b-install.log: No such file or directory
```

### CLI Version

qlty 0.458.0 (d0c76b8)

### Information about Qlty configuration and environment (optional)

```toml
// Add your qlty.toml
```

- OS:

**Comment by brynary** [comment]
Thank you for this report.

We have a bug where we will sometimes suggest looking at a log file which doesn't exist. We will get this fixed up.

The best thing to do to check what is happening is to run the CLI with the `--debug` flag and tail the latest log file in `.qlty/logs/`.

Since there was an error during the Ruby install, I would try this and check that log file:

```
qlty install --debug
``` 

(The `install` subcommand just ensures that all linters and any needed runtimes like Ruby are installed. Otherwise they are lazy-installed when you run `qlty check`.)

Are you still having an issue with the installation of Ruby 3.3.0 via Qlty?

**Comment by brynary** [comment]
One more thing... We have two ways  that the CLI can attempt to install Ruby: by compiling it from source or download a pre-compiled binary install. By default we use the pre-compiled download strategy.

The advantage of downloading pre-compiled binaries is speed, but it can be disabled by setting an env var:

```
export QLTY_FEATURE_RUBY_BINARY_INSTALL=off

qlty install --debug
```

If you are still having issues with Ruby installation, changing this environment variable feature flag would be a good thing to try.

**Comment by nk-ty** [comment]
Thank you for your advice.

An error occurred when I executed the following command.
Also, the log file was not found.

```bash
qlty install --debug
❌ Error installing ruby@3.3.0.

See more: /Users/nk/.qlty/cache/tools/ruby/3.3.0-ce8ae37c7b0b-install.log

Caused by:
Failed to get version for package "ruby": (command sh -c "ruby --version" exited with code 0)

ls /Users/nk/.qlty/cache/tools/ruby/3.3.0-ce8ae37c7b0b-install.log
ls: /Users/nk/.qlty/cache/tools/ruby/3.3.0-ce8ae37c7b0b-install.log: No such file or directory
```

I executed the following command and it was successful.
It took about 30 minutes.

```bash
export QLTY_FEATURE_RUBY_BINARY_INSTALL=off
qlty install --debug
```

Looking forward to the log file fix.
Thank you for your continued support.
